bindings/ruby/lib/rock_gazebo/syskit/configuration_extension.rb

```diff
@@ -23,16 +23,56 @@ def use_sdf_world(*path, world_name: nil)
                 if Conf.sdf.world_file_path
                     raise LoadError, "use_sdf_world already called"
                 elsif Conf.sdf.has_profile_loaded?
-                    raise LoadError, "you need to call #use_sdf_world before require'ing any profile that uses #use_sdf_model"
+                    raise LoadError,
+                          "you need to call #use_sdf_world before require'ing any "\
+                          "profile that uses #use_sdf_model"
                 end
 
                 if Conf.sdf.world_path
                     override_path = Conf.sdf.world_path
-                    Robot.info "world_file_path set on Conf.sdf.world_path with value #{override_path}, overriding the parameter #{File.join(*path)} given to #use_sdf_world"
+                    Robot.info <<~MSG
+                        world_file_path set on Conf.sdf.world_path with value
+                        #{override_path}, overriding the parameter #{File.join(*path)}
+                        given to #use_sdf_world
+                    MSG
                     path = override_path
                 end
 
-                Conf.sdf.load_sdf(*path, world_name: world_name)
+                setup_gazebo_model_path
+                full_path = resolve_world_path(*path)
+                Robot.info "loading world from #{full_path}"
+                Conf.sdf.load_sdf(full_path, world_name: world_name)
+            end
+
+            # Add all models/sdf folders in our dependent bundles
+            def setup_gazebo_model_path(app: Roby.app)
+                roby_paths = Roby.app.find_dirs(
+                    "models", "sdf", all: true, order: :specific_first
+                )
+                Rock::Gazebo.model_path = (roby_paths + Rock::Gazebo.model_path).uniq
```

Comment:
as I see, the gazebo model path will only be overwritten if `use_sdf_world` is called first. Could we get rid of this call dependency?

```rb
 module ConfigurationExtension
       Rock::Gazebo.model_path = setup_gazebo_model_path
       ....
  end
```
```

Reply:
What you propose won't work since we need something to setup the search path, and we'd have a call dependency that we don't control.

In practice, within Syskit, the very first call that handles SDF is `use_sdf_world`. Which is why it is done there.